pkg/provisioner/ironic/ironic.go

```diff
@@ -314,14 +315,23 @@ func (p *ironicProvisioner) createPXEEnabledNodePort(uuid, macAddress string) er
 	return nil
 }
 
+func fmtPreprovExtraKernParams(params string) string {
+	trimmedParams := strings.TrimSpace(params)
+	if trimmedParams == "" {
+		return defaultKernelParam
+	}
+	// spaceing matters for kernel params
+	return defaultKernelParam + " " + params
+}
+
 // configureNode configures Node properties that are not related to any specific provisioning phase.
 // It populates the AutomatedClean field, as well as capabilities and architecture in Properties.
 // It also calls setDeployImage to populate IPA parameters in DriverInfo and
 // checks if the required PreprovisioningImage is provided and ready.
 func (p *ironicProvisioner) configureNode(data provisioner.ManagementAccessData, ironicNode *nodes.Node, bmcAccess bmc.AccessDetails) (result provisioner.Result, err error) {
 	updater := clients.UpdateOptsBuilder(p.log)
 
-	deployImageInfo := setDeployImage(p.config, bmcAccess, data.PreprovisioningImage)
+	deployImageInfo := setDeployImage(p.config, bmcAccess, data.PreprovisioningImage, data)
```

Comment:
nit: either pass `data.PreprovisioningImage` and `data.PreprovisioningExtraKernelParams` or just `data`. Don't pass `data` and one subfield of it.